rclcpp/src/rclcpp/graph_listener.cpp

```diff
@@ -80,7 +85,7 @@ GraphListener::start_if_not_started()
       0,  // number_of_clients
       0,  // number_of_services
       0,  // number_of_events
-      this->parent_context_->get_rcl_context().get(),
+      parent_context_ptr->get_rcl_context().get(),
```

Comment:
The `rcl` context was stored in `interrupt_guard_condition_context_`.
You can directly use `interrupt_guard_condition_context_.get()`, and completely avoid the `parent_context_` lock checking above.

Reply:
Thanks. I will fix it